metadataproxy/roles.py

```diff
@@ -127,6 +127,19 @@ def find_container(ip):
             log.debug(msg.format(_id, ip))
             CONTAINER_MAPPING[ip] = _id
             return c
+        #Not Found ? Let's see if we are running under rancher 1.2+,which uses a label to store the IP
+        _labels = dict(c['Config']['Labels'])
```

Comment:
Is `c['Config']['Labels']` always going to be there? Will it always be castable to a dict? May be good to wrap this in a try/except:

```
try:
    _labels = dict(c['Config']['Labels'])
except (KeyError, ValueError):
    _labels = {}
```